optuna/storages/_rdb/alembic/versions/v3.0.0.c.py

```diff
@@ -94,20 +94,30 @@ def upgrade():
 
     session = orm.Session(bind=bind)
     try:
-        records = session.query(IntermediateValueModel).all()
+        records = (
+            session.query(IntermediateValueModel)
+            .filter(
+                sa.or_(
+                    IntermediateValueModel.intermediate_value > 1e16,
+                    IntermediateValueModel.intermediate_value < -1e16,
+                    IntermediateValueModel.intermediate_value.is_(None),
+                )
+            )
+            .all()
+        )
         mapping = []
         for r in records:
             value: float
-            if np.isclose(r.intermediate_value, RDB_MAX_FLOAT) or np.isposinf(
+            if r.intermediate_value is None or np.isnan(r.intermediate_value):
+                value = float("nan")
```

Comment:
@contramundum53 This line does not distinguish `None` and `nan`. I think this is reasonable according to [this comment](https://github.com/optuna/optuna/pull/4020/files#r981124322). I don't remember the discussion in ##3603, but do you have any concerns about this change?

Reply:
One concern is https://github.com/optuna/optuna/pull/4020/files#r981834790, where half-migrated `inf` values would be considered as `nan` values. However, we decided not to consider the case where the batch operation fails halfway, so I don't think that is a problem now.

Reply:
Thank you for your comment. That makes sense.